test/e2e/deploy/deploy.go

```diff
@@ -181,6 +182,29 @@ func WaitForPMEMDriver(c *Cluster, name, namespace string) (metricsURL string) {
 	}
 }
 
+// CheckPMEMDriver does some sanity checks for a running deployment.
+func CheckPMEMDriver(c *Cluster, deployment *Deployment) {
+	pods, err := c.cs.CoreV1().Pods(deployment.Namespace).List(context.Background(),
+		metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("%s in (%s)", deploymentLabel, deployment.Name),
+		},
+	)
+	framework.ExpectNoError(err, "list PMEM-CSI pods")
+	gomega.Expect(len(pods.Items)).Should(gomega.BeNumerically(">", 0), "should have PMEM-CSI pods")
+	for _, pod := range pods.Items {
+		for _, containerStatus := range pod.Status.ContainerStatuses {
+			if containerStatus.RestartCount > 0 {
```

Comment:
Or we fix this problem. If it's because of a timeout, then we may be able to increase that timeout.

Reply:
I think it's much simpler: that happened during the reboot tests, so of course all containers were restarted...

I'm not sure yet how to deal with this. The check is useful because it highlights a problem that went unnoticed without it for a while. Perhaps clean up after the reboot tests by killing all pods?

Reply:
In this case, checking if the pod container restart count == node reboot count might work.

Reply:
I've implemented that approach.

Reply:
@avalluri okay now? Tests have passed.

Reply:
Yes, we can merge.